### ISSUE TITLE
Algorythm Request Apple File Vault 2

### ISSUE THREAD

**Issue by IncognitoEntity** [hashcat/hashcat#1028]
I'm just wondering if it would be possible to add GPU support for apple file vault 2 within hashcat?  

I recently posted about IOS backups and this implementation has been a great success.  This is why I am wondering how possible it would be to add FileVault 2 to hashcat.  Due to Apple Macs being encrypted by default now there is a huge need within my area to be able to use a software to try to gain access to the computers.  

If I can help in any way by providing the first sectors of a FileVault 2 image let me know.  There is only one software (Passware) that I know about that can do filevault 2 however again I feel hashcat developers would be able to provide a far better faster solution using GPU's

Thanks in advance 

Curtis

**Comment by hcbob** [comment]
I'd like to add my support for this - I also come across a lot of FV2 encrypted Macs and having support for it in Hashcat would be great. I could also provide some volume headers and EncryptedRoot.plist.wipekey combos for research if needed.

Many thanks!

**Comment by Banaanhangwagen** [comment]
I am also interest in this new algorithm.
With [libfvde/fdvetools](https://github.com/libyal/libfvde/tree/master/fvdetools) it is possible to extract the hash-value. 
Extra info can be found in this [documentation](https://github.com/libyal/libfvde/blob/master/documentation/FileVault%20Drive%20Encryption%20(FVDE).asciidoc)

Apparently, [jtr ](https://github.com/magnumripper/JohnTheRipper/pull/2436/files) supports it also. 
Thanks!

**Comment by DoZ10** [comment]
According to @Banaanhangwagen Doc link, this looks like AES-XTS 128 bits.

I believe there is another Issue opened for this mode (Bitlocker for Windows 10).

KDF is PBKDFv2 with salted SHA-256 (which is already implemented).

This might be simple to implement once AES-XTS is done. 

**EDIT**: Truecrypt and Veracrypt seem to implement XTS... perhaps this could be feasible.

**Comment by Banaanhangwagen** [comment]
A little bump.
Some documentation: http://www.cl.cam.ac.uk/%7Eosc22/docs/slides_fv2_ifip_2013.pdf

**Comment by mneitsabes** [comment]
I am also very interested in FileVault support. I can also provide examples if needed or perform tests.

**Comment by jsteube** [comment]
Support added with https://github.com/hashcat/hashcat/commit/c28fdf7f44bf5d39f9078f6d82b52bdb052125e3
